presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -164,29 +170,95 @@ public OrcSelectiveRecordReader(
             this.constantValues[zeroBasedIndices.get(entry.getKey())] = entry.getValue();
         }
 
+        Map<Integer, Map<Subfield, TupleDomainFilter>> filtersWithNewIndices = filters
```

Comment:
I think you can change `columnsWithFilters` to be a map from column index to filter score and avoid introducing new variable.

```
        columnsWithFilters = filters.entrySet().stream()
                .collect(toImmutableMap(entry -> zeroBasedIndices.get(entry.getKey()), entry -> scoreFilter(entry.getValue())));
        ...
        streamReaderOrder = orderStreamReaders(columnTypes.keySet().stream().filter(index -> this.constantValues[index] == null).collect(toImmutableSet()), columnsWithFilters, filterFunctionInputs);
```

Reply:
This makes sense. But with one caveat  -- what do we do with columns that have more than one filter applied to them?

I am putting in logic to deal with that. What do you think -- should we go for the MAX of the two scores, or the MIN?

That is, let's say a column has both an integer equality and an integer inequality applied to it. Should the column end up with a score of 10 (for equality) or 50 (for inequality?)

Reply:
(This comes from the idea that we are scoring filters, but reading one *column* at a time, and applying all filters on that column at once.)